eks-hybrid-ssm-setup.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+exec >&2
+set -eux -o pipefail
+
+HOST_ROOTFS="/.bottlerocket/rootfs"
+SSM_AGENT_PERSISTENT_STATE_DIR="${HOST_ROOTFS}/local/host-containers/control/ssm"
+SSM_AGENT_REGISTRATION="${SSM_AGENT_PERSISTENT_STATE_DIR}/registration"
+mkdir -p "${SSM_AGENT_PERSISTENT_STATE_DIR}"
+ENABLE_CREDENTIALS_FILE="false"
+DRY_RUN="false"
+SSM_ACTIVATION_ID_REGEX="^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"
+
+set +x
+for opt in "$@"; do
+    optarg="$(expr "${opt}" : '[^ =]*[= ]\(.*\)')"
+    case "${opt}" in
+       --region=*) AWS_REGION="${optarg}" ;;
+       --activation-code=*) SSM_ACTIVATION_CODE="${optarg}" ;;
+       --activation-id=*) SSM_ACTIVATION_ID="${optarg}" ;;
+       --enable-credentials-file=*) ENABLE_CREDENTIALS_FILE="${optarg}" ;;
+        --dry-run=*) DRY_RUN="${optarg}" ;;
+    esac
+done
+set -x
+
+validate_activation_id() {
+    if ! [[ $SSM_ACTIVATION_ID =~ ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$ ]]; then
+        echo "Error: Invalid activation ID format" >&2
+        exit 1
+    fi
+}
+
+extract_ssm_agent() {
+  pushd / >/dev/null 2>&1
+  dnf -yq install 'dnf-command(download)' cpio rsync
+  dnf download amazon-ssm-agent
+  rpm2cpio amazon-ssm-agent-*.rpm | cpio -idmu '*bin/amazon-ssm-agent'
+  popd >/dev/null 2>&1
+}
```

Comment:
We could move these steps to the Dockerfile so they happen at build time, and `amazon-ssm-agent` could be copied to `/usr/local/bin`.